From 2ad89e55ebd860b1444213f9b76ce5d0b909b8f8 Mon Sep 17 00:00:00 2001
From: Marta Kuczora <kuczoram@cloudera.com>
Date: Tue, 8 Aug 2017 15:07:30 +0200
Subject: [PATCH 1232/1363] CLOUDERA-BUILD: CDH-57609: Fixing the TestHive
 unit test The test failed due to the side effect
 of HDFS-8312.

Change-Id: I749a8f5c16af569d115509173ee16f9d1cf8e8f1
---
 .../java/org/apache/hadoop/fs/ProxyFileSystem.java |    8 ++++++++
 .../org/apache/hadoop/fs/ProxyLocalFileSystem.java |    6 ++++++
 2 files changed, 14 insertions(+)

diff --git a/shims/common/src/main/java/org/apache/hadoop/fs/ProxyFileSystem.java b/shims/common/src/main/java/org/apache/hadoop/fs/ProxyFileSystem.java
index 2c37a51..14c29bc 100644
--- a/shims/common/src/main/java/org/apache/hadoop/fs/ProxyFileSystem.java
+++ b/shims/common/src/main/java/org/apache/hadoop/fs/ProxyFileSystem.java
@@ -23,6 +23,7 @@
 import java.net.URISyntaxException;
 
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Options.Rename;
 import org.apache.hadoop.fs.permission.FsPermission;
 import org.apache.hadoop.util.Progressable;
 import org.apache.hadoop.util.Shell;
@@ -181,6 +182,13 @@ public boolean rename(Path src, Path dst) throws IOException {
   }
 
   @Override
+  public void rename(Path src, Path dst, Rename... options) throws IOException {
+    Path dest = swizzleParamPath(dst);
+    Path source = swizzleParamPath(src);
+    super.rename(source, dest, options);
+  }
+
+  @Override
   public boolean delete(Path f, boolean recursive) throws IOException {
     return super.delete(swizzleParamPath(f), recursive);
   }
diff --git a/shims/common/src/main/java/org/apache/hadoop/fs/ProxyLocalFileSystem.java b/shims/common/src/main/java/org/apache/hadoop/fs/ProxyLocalFileSystem.java
index 228a972..9da7427 100644
--- a/shims/common/src/main/java/org/apache/hadoop/fs/ProxyLocalFileSystem.java
+++ b/shims/common/src/main/java/org/apache/hadoop/fs/ProxyLocalFileSystem.java
@@ -22,6 +22,7 @@
 import java.net.URI;
 
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Options.Rename;
 import org.apache.hadoop.util.Shell;
 import org.apache.hadoop.hive.shims.ShimLoader;
 import org.apache.hadoop.hive.shims.HadoopShims;
@@ -69,4 +70,9 @@ public void initialize(URI name, Configuration conf) throws IOException {
 
     fs.initialize(name, conf);
   }
+
+  @Override
+  public void rename(Path src, Path dst, Rename... options) throws IOException {
+    fs.rename(src, dst, options);
+  }
 }
-- 
1.7.9.5

